trunk/SUAVE/Methods/Constraint_Analysis/compute_turn_constraint.py

```diff
@@ -0,0 +1,129 @@
+
+
+# package imports
+import numpy as np
+
+# ----------------------------------------------------------------------
+#  Compute turn constraint
+# ----------------------------------------------------------------------
+
+## @ingroup Methods-Constraint_Analysis
+def compute_turn_constraint(vehicle):
+    
+    """Calculate thrust-to-weight ratios for the turn maneuver
+
+        Assumptions:
+            Minimum drag coefficient is independent on the altitude and airspeed
+
+        Source:
+            S. Gudmundsson "General Aviation aircraft Design. Applied methods and procedures", Butterworth-Heinemann (6 Nov. 2013), ISBN - 0123973082
+
+        Inputs:
+            self.aerodynamics.cd_min_clean      [Unitless]
+                 engine.type                    string
+                 turn.angle                     [radians]
+                      altitude                  [m]
+                      delta_ISA                 [K]
+                      airspeed                  [m/s]
+                      specific_energy
+                      thrust_fraction           [Unitless]
+                 geometry.aspect_ratio          [Unitless]
+                 wing_loading                   [N/m**2]
+
+        Outputs:
+            constraints.T_W                     [Unitless]
+                        W_S                     [N/m**2]
+                        
+        Properties Used:
+    """         
+
+    # ==============================================
+    # Unpack inputs
+    # ==============================================
+    cd_min    = vehicle.constraints.aerodynamics.cd_min_clean 
+    eng_type  = vehicle.constraints.engine.type
+    phi       = vehicle.constraints.analyses.turn.angle         
+    altitude  = vehicle.constraints.analyses.turn.altitude        
+    delta_ISA = vehicle.constraints.analyses.turn.delta_ISA       
+    M         = vehicle.constraints.analyses.turn.mach       
+    Ps        = vehicle.constraints.analyses.turn.specific_energy 
+    throttle  = vehicle.constraints.analyses.turn.thrust_fraction   
+    AR        = vehicle.wings['main_wing'].aspect_ratio
+    W_S       = vehicle.constraints.wing_loading
+
+
+    # Determine atmospheric properties at the altitude
+    planet      = SUAVE.Analyses.Planets.Planet()
+    atmosphere  = SUAVE.Analyses.Atmospheric.US_Standard_1976()
+    atmosphere.features.planet = planet.features
+
+    atmo_values = atmosphere.compute_values(altitude,delta_ISA)
+    rho         = atmo_values.density[0,0]
+    Vturn       = M * atmo_values.speed_of_sound[0,0]
+    q           = 0.5*rho*Vturn**2  
+
+    T_W = np.zeros(len(W_S))
+    if eng_type != ('turbofan' or 'Turbofan') and eng_type != ('turbojet' or 'Turbojet'):
+        P_W  = np.zeros(len(W_S))
+        etap = vehicle.constraints.propeller.turn_efficiency
+        if etap == 0:
+            raise ValueError('Warning: Set the propeller efficiency during turn')
+
+    for i in range(len(W_S)):
```

Comment:
2 things I see.

First, this section of code is used repeatedly in several scripts. This indicates this could be functionalized.

Second, I believe you could vectorize this for loop.